timed/src/lib.rs

```diff
@@ -127,69 +124,89 @@ impl Trace {
         }
     }
 
-    pub fn dump(id: &str, options: &TraceOptions) {
-        if options.chrome_trace.is_none() && options.statistics.is_none() { return; }
+    pub fn new(id: &str, options: TraceOptions) -> Trace {
+        let trace = Trace {
+            id: id.into(),
+            options,
+        };
+        Self::register(&trace.id);
+        trace
+    }
+
+    pub fn finish(self) {
+        self.dump();
+    }
```

Comment:
When I removed the `init_tracing!` macro, I replaced it with `let _ = TraceOptions...build_named("Test")` and got empty tracings, this was due to the drop that was called before it should have been called (due to my changes).

This made me think that the drop behaviour was too implicit, that's why I replaced it. Please let me know if you dissagree

Reply:
yep, `_` has a special treatment that it drops immediately. that's why maybe it helps when it's hidden behind a macro. However, calling it `_trace` will make it drop at the end of the block, so that should work. 

I have mixed feelings, I want a clean api and I think that combining a macro with a pseudo builder pattern is unidiomatic, a bit clunky. By any means, this is not bad feedback, I am just unsure. If we can keep macros simple and clean, all good, but if not, i'd rather keep it simple.

I do think that if you make users init the variables the normal way, most of them will use `_` as a name so that makes the api not intuitive, users won't know they can rely on Drop.

I think `finish` is a good choice, I don't have a better idea. Forcing users to call finish will ensure they name their variables properly and therefore it doesn't drop prematurely.  

Good job with this, really good stuff, thanks a lot.